js/bitget.js

```diff
@@ -2459,6 +2470,68 @@ module.exports = class bitget extends Exchange {
         };
     }
 
+    async modifyMarginHelper (symbol, amount, type, params = {}) {
+        await this.loadMarkets ();
+        const holdSide = this.safeString (params, 'holdSide');
+        const market = this.market (symbol);
+        const marginCoin = (market['linear']) ? market['quote'] : market['base'];
+        const request = {
+            'symbol': market['id'],
+            'marginCoin': marginCoin,
+            'amount': this.amountToPrecision (symbol, amount), // positive value for adding margin, negative for reducing
+            'holdSide': holdSide, // long or short
+        };
+        params = this.omit (params, 'holdSide');
+        const response = await this.privateMixPostAccountSetMargin (this.extend (request, params));
+        //
+        //     {
+        //         "code": "00000",
+        //         "msg": "success",
+        //         "requestTime": 1652483636792,
+        //         "data": {
+        //             "result": true
+        //         }
+        //     }
+        //
+        return this.extend (this.parseModifyMargin (response, market), {
+            'amount': this.parseNumber (amount),
+            'type': type,
+        });
+    }
+
+    parseModifyMargin (data, market = undefined) {
+        const errorCode = this.safeString (data, 'code');
+        const status = (errorCode === '00000') ? 'ok' : 'failed';
+        const code = (market['linear']) ? market['quote'] : market['base'];
+        return {
+            'info': data,
+            'type': undefined,
+            'amount': undefined,
+            'code': code,
+            'symbol': market['symbol'],
+            'status': status,
+        };
+    }
+
+    async reduceMargin (symbol, amount, params = {}) {
+        if (amount > 0) {
```

Comment:
@kroitor @carlosmiei

Should we use string math for this?

Reply:
@samgermain I don't think so because this is a very simple integer comparison therefore, we're not losing any information like in the floating point rounding, moreover, we already receive the amount as an integer, so we would need to convert it which is not worth it IMO for the reason mentioned above